index.js

```diff
@@ -1,97 +1,226 @@
+                handleLocalError(name, responseDate);
+            } else if (response.statusCode == 200) {
+                handleSuccessStatus(name, params.url, responseDate);
             } else {
-                if (typeof notReachable[name] == "undefined") { // first time
-                    notReachable[name] = "true";
-                    var message = name + " " + emoji.get('hankey') + " " + getFormattedDate();
-                    console.log("first time :" + message);
-
-                } else if (notReachable[name] == "true") { // secondtime
-                    notReachable[name] = "notagain"; // we don't want second notification
-                    var message = name + " " + emoji.get('hankey') + " " + getFormattedDate();
-                    console.log(message);
-                    send(message);
-                }
+                handleDownStatus(name, params.url, responseDate, params.mentionsWhenDown);
             }
         });
-    }, interval);
+    }, params.interval);
+}
+
+function handleSuccessStatus(urlName, urlValue, responseDate) {
+    if (urlContainer[urlName] === undefined) {
+        const message = "First check of this session... Endpoint is running! :beer:";
+        send(message);
+        console.log(`${getFormattedDate(responseDate)} - ${message}`);
+    } else {
+        if (urlErrorContainer[urlName] !== undefined) {
+            restoreLocalErrorAndSend(responseDate, urlName);
+        }
+
+        if (urlContainer[urlName] !== UrlStatus.Up) {
+            const message = `:globe_with_meridians: ${urlName} (${urlValue}) is up and running!`;
+            send(message);
+            console.log(`${getFormattedDate(responseDate)} - ${message}`);
+        }
+    }
+
+    urlContainer[urlName] = UrlStatus.Up;
+}
+
+function handleLocalError(urlName, responseDate) {
+    console.log('Local error, couldn\'t reach endpoint.');
+
+    if (urlErrorContainer[urlName] === undefined) {
+        urlErrorContainer[urlName] = responseDate;
+    }
+}
+
+function handleDownStatus(urlName, urlValue, responseDate, mentionsWhenDown) {
+    const trailingMentions = buildMentionsForDownStatus(mentionsWhenDown);
+    const message = `:warning: Oh no! It looks like ${urlName} (${urlValue}) is down.\n${trailingMentions}`;
+    console.log(`${getFormattedDate(responseDate)} - ${message}`);
+
+    if (urlContainer[urlName] === undefined) {
+        send("First check of this session... Endpoint didn't respond :confused:");
+        urlContainer[urlName] = UrlStatus.Pending;
+        return;
+    }
+
+    if (urlErrorContainer[urlName] !== undefined) {
+        restoreLocalErrorAndSend(responseDate, urlName);
+    }
+
+    if (urlContainer[urlName] === UrlStatus.Up) {
+        // First fail! Try again, maybe this was temporary.
+        urlContainer[urlName] = UrlStatus.Pending;
+    } else if (urlContainer[urlName] === UrlStatus.Pending) {
+        // Two fails in a row! Send message, the endpoint is down.
```

Comment:
hum... isn't it better to replace these status with a counter? then you can even say how many failures in a row

Reply:
We only send a message once, when the URL is confirmed to be down (2 fails in a row), and then another one when it comes back up. Not sure where we'd fit the counter in the message

Reply:
what I mean is more in the `UrlStatus` to be replaced with a failures counter. Then it's easy to print the counter in the message IF NEEDED, also able to make it configurable "how many failures before I call it a failure", right now it's 2, but eventually you want 1 or 10.